### PR TITLE
Continue building action

### PR DESCRIPTION
**Pull request by geramirez**

### CODE REVIEW

**Comment by geramirez** [comment]
@dmilburn I just finished adding the error logs when the template fails while polling... I think we have an issue..

How do we get the prebuild template guid? Can we add it to the status response?

File: cache-codespace.sh [hunk 1]
```diff
@@ -45,10 +70,18 @@ for region in $INPUT_REGIONS; do
...
 )
-  response=$(curl -X POST "${GITHUB_API_URL}/vscs_internal/codespaces/repository/${GITHUB_REPOSITORY}/prebuild/templates" \
+  http_code=$(curl -X POST "${GITHUB_API_URL}/vscs_internal/codespaces/repository/${GITHUB_REPOSITORY}/prebuild/templates" \
     -H "Content-Type: application/json; charset=utf-8" \
     -H "Authorization: token $GITHUB_TOKEN" \
-    -d "$body")
-  job_id=$(echo $response | jq -r '.job_status_id') 
-  poll_status $job_id
+    -d "$body" \
+    -o response_body.txt \
```
Comment (dmilburn) [inline cache-codespace.sh:77]: Why save this to a file?

File: cache-codespace.sh [hunk 2]
```diff
@@ -7,24 +7,49 @@ get_status () {
...
     poll_status "$job_id" $(($attempt+1))
   fi
 }
 
+handle_error_message() {
+  local error_message="$1"
+  >&2 echo "*************************"
+  >&2 echo "Error message from server"
+  >&2 echo "$error_message"
+  >&2 echo "*************************"
```
Comment (dmilburn) [inline cache-codespace.sh:50]: Is your sense that this error message will be user friendly?

File: cache-codespace.sh [hunk 3]
```diff
@@ -7,24 +7,49 @@ get_status () {
...
+  error_logs_available="$(echo $status_data | jq -r '.error_logs_available')"
+
+  if [[ "$error_logs_available" == "true" ]]; then
+    guid="$(echo $status_data | jq -r '.guid')"
+    build_logs=$(curl "${GITHUB_API_URL}/vscs_internal/codespaces/repository/${GITHUB_REPOSITORY}/prebuilds/environments/$guid/logs" \
+      -H "Content-Type: application/json; charset=utf-8" \
+      -H "Authorization: token $GITHUB_TOKEN")
+    handle_error_message "$build_logs"
+  else
+    handle_error_message "$(echo $status_data | jq -r '.message')"
```
Comment (dmilburn) [inline cache-codespace.sh:22]: There should be a `message` if `!error_log_available` but to make it more resilient to future changes, maybe it'd be worthwhile to add a blanket error message if both neither error_logs_available nor message exist

File: test/action_test.rb [hunk 4]
```diff
@@ -289,25 +379,82 @@ def with_fake_api_server_running(job_ids, status_responses)
...
+  end
+
+  def body
+    {state: state, message: message, error_logs_available: error_logs_available, guid: guid}.to_json
+  end
+
+  def status
+    200
+  end
+end
```
Comment (dmilburn) [inline test/action_test.rb:446]: Love this

File: cache-codespace.sh (hunk 1 above)
Comment (geramirez) [reply cache-codespace.sh:77]: I wanted the status code and variable to go to different places. There is probably a better way to do this. Maybe we can figure it out during our pairing time.

File: cache-codespace.sh (hunk 2 above)
Comment (geramirez) [reply cache-codespace.sh:50]: We could probably create a better output. I put it into a function so we can change it quickly moving forward.

**Review by dmilburn — APPROVED** [review]
LG!

**Pull request merged**